server/core/dcb.cc

```diff
@@ -508,6 +512,10 @@ dcb_connect(SERVER *server, MXS_SESSION *session, const char *protocol)
         dcb_free_all_memory(dcb);
         return NULL;
     }
+    
+    /* Register throttling callbacks */
+    dcb_add_callback(dcb, DCB_REASON_HIGH_WATER, session_upstream_throttle_callback, NULL);
```

Comment:
The callbacks should only be added if the throttling is enabled.

Reply:
```
#define DCB_BELOW_LOW_WATER(x)          ((x)->low_water && (x)->writeqlen < (x)->low_water)
#define DCB_ABOVE_HIGH_WATER(x)         ((x)->high_water && (x)->writeqlen > (x)->high_water)


if (DCB_ABOVE_HIGH_WATER(dcb) && !dcb->high_water_has_reached)
{
        dcb_call_callback(dcb, DCB_REASON_HIGH_WATER);
        ....
}
```

Since throttle is disable when high_water == 0, so I think it is safe to add throttling callbacks

Reply:
It is safe to add them but it will cause an extra memory allocation. For this reason the callbacks should not be added unless the throttling is enabled.